### PR TITLE
Add SpacialScore

### PR DESCRIPTION
**Pull request by apahl**
Implementation of the SpacialScore for the RDKit, based on the original implementation by @frog2000.  
As part of the Hackathon of the 2023 UGM the code was adapted to fit into the RDKit Descriptor module by: @jose-manuel, Arsenio Cruz, @vincenzo-palmacci and myself.

Many thanks to @rvianello for his help in setting everything up!

### CODE REVIEW

**Review by DavidACosgrove — COMMENTED** [review]
If it’s not too much of a pain at this point, can we call it SpatialScore?

**Comment by frog2000** [comment]
> If it’s not too much of a pain at this point, can we call it SpatialScore?

One of the coauthors here. No ;-b

File: rdkit/Chem/SpacialScore.py [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+s = Stereoisomeric term
+r = Non-aromatic ring term
+n = Number of heavy atom neighbors
+a = Total number of heavy atoms in the molecule
+
+The SPS function in this module takes a mol object and returns either the absolute score (normalize=False) or the score normalized by the number of heavy atoms (normalize=True (default)).
+
+The original code implementation can be found at: https://github.com/frog2000/Spacial-Score/blob/main/spacial_score.py
+
+[1] Krzyzanowski, A., Pahl, A., Grigalunas, M., Waldmann, H. Spacial Score-A Comprehensive Topological Indicator for Small Molecule Complexity. [https://doi.org/10.26434/chemrxiv-2023-nd1ll]
```
Comment (frog2000) [inline rdkit/Chem/SpacialScore.py:57]: It's probably better to use this DOI from J. Med. Chem. instead: https://doi.org/10.1021/acs.jmedchem.3c00689

Comment (apahl) [reply rdkit/Chem/SpacialScore.py]: Good catch. I did not notice that the link was still for the preprint. Added a commit to fix the citation.  
BTW, the Github handle for Arsenio is: @As337N.

**Review by greglandrum — CHANGES_REQUESTED** [review]
I made some suggested changes in a PR against your branch.
If you agree with those changes and merge the PR, the changes will automatically show up here.

Let me know (in the PR or here) if you have questions or concerns about any of my suggestions.

**Review by greglandrum — APPROVED** [review]
LGTM

**Comment by greglandrum** [comment]
#hackathon

**Pull request merged**